registrations/test_metrics.py

```diff
@@ -23,6 +30,474 @@ def test_generate_metric(self):
+
+        self.create_registration_on(
+            datetime(2016, 10, 14), source, operator_id='id1')  # Before
+        self.create_registration_on(
+            datetime(2016, 10, 15), source, operator_id='id1')  # On
+        self.create_registration_on(
+            datetime(2016, 10, 20), source, operator_id='id1')  # During
+        self.create_registration_on(
+            datetime(2016, 10, 20), source, operator_id='id2')  # Wrong type
+        self.create_registration_on(
+            datetime(2016, 10, 25), source, operator_id='id1')  # On
+        self.create_registration_on(
+            datetime(2016, 10, 26), source, operator_id='id1')  # After
+
+        reg_count = MetricGenerator().registrations_role_sum(
+            'role1', start, end)
+        self.assertEqual(reg_count, 2)
+
+    @responses.activate
+    def test_registrations_role_total_last(self):
+        """
+        Should return the amount of registrations up to the end of the
+        timeframe.
+        """
+        user = User.objects.create(username='user1')
+        source = Source.objects.create(
+            name='TestSource', authority='hw_full', user=user)
+
+        url = 'http://localhost:8001/api/v1/identities/search/?' \
+              'details__role=role1'
+        responses.add_callback(
+            responses.GET, url, callback=self.identity_search_callback,
+            content_type="application/json", match_querystring=True)
+
+        start = datetime(2016, 10, 15)
+        end = datetime(2016, 10, 25)
+
+        self.create_registration_on(
+            datetime(2016, 10, 14), source, operator_id='id1')  # Before
+        self.create_registration_on(
+            datetime(2016, 10, 20), source, operator_id='id1')  # During
+        self.create_registration_on(
+            datetime(2016, 10, 20), source, operator_id='id2')  # Wrong type
+        self.create_registration_on(
+            datetime(2016, 10, 25), source, operator_id='id1')  # On
+        self.create_registration_on(
+            datetime(2016, 10, 26), source, operator_id='id1')  # After
+
+        reg_count = MetricGenerator().registrations_role_total_last(
+            'role1', start, end)
+        self.assertEqual(reg_count, 3)
+
+    def test_registrations_source_sum(self):
+        """
+        Should return the amount of registrations in the given timeframe for
+        a specific source.
+
+        Only one of the borders of the timeframe should be included, to avoid
+        duplication.
+        """
+        user1 = User.objects.create(username='user1')
+        user2 = User.objects.create(username='user2')
+        source1 = Source.objects.create(
+            name='TestSource1', authority='hw_full', user=user1)
+        source2 = Source.objects.create(
+            name='TestSource2', authority='hw_full', user=user2)
+
+        url = 'http://localhost:8001/api/v1/identities/search/?' \
+              'details__state=state1'
+        responses.add_callback(
```

Comment:
Nope, the test doesn't even have the responses decorator so this does nothing. I'll remove it.